### ISSUE TITLE
Inconsistent :setBlendFunc(...) function binding for Sprite and LabelBMFont classes

### ISSUE THREAD

**Issue by Yamomota** [cocos2d/cocos2d-x#9820]
Hi friends,

**First of all I would like to thank you for your great work. 8)**
I've found the following issues (it was in the previous releases (3.3rc0,3.3rc1,3.3rc2) as well) in the latest 3.3 release:
 :setBlendFunc(...) is bound to lua for Sprite class and LabelBMFont one in different way. The following code fragment explains the difference:

``` lua
local n = cc.Sprite:create('texture.png')
n:setBlendFunc(gl.ONE, gl.ONE_MINUS_SRC_ALPHA)  -- uses two value as in all exaples

local l = cc.LabelBMFont:create('just a text', 'font.fnt')
l:setBlendFunc(gl.ONE, gl.ONE_MINUS_SRC_ALPHA) -- **ERROR**: As this function expects only one argument by some reason 8(((
l:setBlendFunc({gl.ONE, gl.ONE_MINUS_SRC_ALPHA}) -- it **WORKS** if pass blending operands are passed inside a table more like it's done in C++
```

Of course this issue is minor as it lives for such long time but any way. 8)
Thank you and have a nice day.
Y

**Comment by Yamomota** [comment]
Hi all again,

Additionally, if the blending operands are passed via table (the code below just in case):

``` lua
local l = cc.LabelBMFont:create('just a text', 'font.fnt')
l:setBlendFunc({gl.ONE, gl.ONE_MINUS_SRC_ALPHA})
```

 it causes **OpenGL error 0x500 in ..\render\CCTextureAtlas.cpp cocos2d::TExtureAtlas::drawNumberOfQuads 689** on each rendered frame. Of course it's going to be caused of some improper specified OpenGL parameters (blending operators). 
**Nevertheless everything visually looks good. There is a rendered text on the screen. 8)**

Thank you.
Have a nice day,
Y

**Comment by samuele3hu** [comment]
Thank you @Yamomota .This problem does exists for a long time. At the begin I think using two parameters for setBlendFunc would better,It can reduce a table conversion.But, this function can be auto-binding by adding the convertible functions and wouldn't  be called frequently. So i think i should modify the bindings-generator and add a convertible functions to solve the problems that parameters are not unified. After the modification, we call this function by passing a table parameter.At the same time,we also add deprecated function to keep compatibility.

At current time，setBlendFunc of LabelBMFont was invalid because the binding codes didn't convert the passing table.

**Comment by minggo** [comment]
Fixed in #9983.
